realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -23,14 +23,68 @@
 
 /**
  * A MutableRealmInteger is a mutable, {@link java.lang.Long}-like numeric quantity.
- * MutableRealmInteger implementations are not thread safe.  Like all Realm objects,
- * managed MutableRealmIntegers may not be moved across threads.  Unmanaged MutableRealmIntegers
- * may be moved across threads but require safe publication.
+ * It behaves almost exactly as a reference to a {@link java.lang.Long}. More specifically:
+ * <ul>
+ * <li>The MutableRealmInteger may have the value {@code null}.</li>
+ * <li>The {@code .equals} operator compares the contained Long values. null-valued MutableRealmInteger are {@code .equals}</li>
+ * <li>The {@code .compareTo} operator compares the contained Long values.  It considers {@code null} &lt; any value.</li>
```

Comment:
I mean to say that null == null, null < x, for any x.  Suggest an alternative wording?